lib/sshutils/sftp/sftp.go

```diff
@@ -667,3 +666,15 @@ func NewProgressBar(size int64, desc string, writer io.Writer) *progressbar.Prog
 		progressbar.OptionSetRenderBlankState(true),
 	)
 }
+
+// NonRecursiveDirectoryTransferError is returned when an attempt is made
+// to download a directory without providing the recursive option.
+// It's used to distinguish this specific situation in clients which
+// do not support the recursive option.
+type NonRecursiveDirectoryTransferError struct {
+	Path string
+}
+
+func (n *NonRecursiveDirectoryTransferError) Error() string {
+	return fmt.Sprintf("%q is a directory, but the recursive option was not passed", n.Path)
```

Comment:
I'm not sure about the wording style - generally "inner" error messages in Go are a bit terser than this, but this will also double as a user-facing message, right?

Reply:
Yes. I forgot to include screenshots which I've now just added, it show also answer your other question.